internal/pkg/util/fs/helper.go

```diff
@@ -134,34 +134,53 @@ func RootDir(path string) string {
 	return p
 }
 
-// EvalRelative evaluates symlinks in path relative to root path. This
-// function doesn't return error but always returns an evaluated path.
+// EvalRelative evaluates symlinks in path relative to root path, it returns
+// a path as if it was evaluated from chroot. This function always returns
+// an absolute path.
 func EvalRelative(path string, root string) string {
-	splitted := strings.Split(filepath.Clean(path), string(os.PathSeparator))
-	dest := string(os.PathSeparator)
-
-	for i := 1; i < len(splitted); i++ {
-		s := splitted[i : i+1][0]
-		dest = filepath.Join(dest, s)
-
-		if s != "" {
-			rootDestPath := filepath.Join(root, dest)
-			for {
-				target, err := filepath.EvalSymlinks(rootDestPath)
-				if err != nil {
-					break
-				}
-				if !strings.HasPrefix(target, root) {
-					rootDestPath = filepath.Join(root, target)
-					continue
+	cleanRoot := filepath.Clean(root)
+	cleanPath := filepath.Clean(path)
+
+	if cleanPath == "" {
+		return cleanRoot
+	}
+
+	sep := string(os.PathSeparator)
+	splitted := strings.Split(cleanPath, sep)
+
+	i := 1
+	if !filepath.IsAbs(cleanPath) {
+		i = 0
+	}
+
+	dest := cleanRoot
+
+	for ; i < len(splitted); i++ {
+		dest = filepath.Join(dest, splitted[i])
+
+		fi, err := os.Lstat(dest)
+		if err != nil {
+			continue
+		}
+
+		if fi.Mode()&os.ModeSymlink != 0 {
```

Comment:
just to avoid indenting the code too much, I would do:

```
if fi.Mode() & os.ModeSymlink == 0 {
    // not a symlink, continue
    continue
}
```